cuegui/cuegui/AbstractTreeWidget.py

```diff
@@ -346,7 +346,10 @@ def _removeItem(self, item):
         self.takeTopLevelItem(self.indexOfTopLevelItem(item))
         objectClass = item.rpcObject.__class__.__name__
         objectId = item.rpcObject.id()
-        del self._items['{}.{}'.format(objectClass, objectId)]
+
+        # Check if object ID exists before delete it
+        if objectId in self._items:
+            del self._items['{}.{}'.format(objectClass, objectId)]
```

Comment:
I find this a bit confusing. If we're getting a `KeyError`, that means we're trying to delete an item from the dict that doesn't exist, right?

However in this change, the key we're checking (`objectId`) is different than the one we're trying to delete (`{objectClass}.{objectId}`).

Should it actually be something like:

```
objectKey = '{}.{}'.format(objectClass, objectId)
if objectKey in self._items:
  del self._items[objectKey]
```

An alternative could be to just catch `KeyError` and ignore it, since it means the deletion already happened.